sm/tedge_agent/src/state.rs

```diff
@@ -61,8 +60,8 @@ impl StateRepository<State> for AgentStateRepository {
 }
 
 impl AgentStateRepository {
-    pub fn new(config_location: &TEdgeConfigLocation) -> Self {
-        let mut state_repo_root = config_location.tedge_config_root_path.clone();
+    pub fn new(tedge_root: PathBuf) -> Self {
+        let mut state_repo_root = tedge_root.clone();
```

Comment:
It was done in an attempt to make the agent modules independent of the `TEdgeConfig` and keep that config dependency limited to the `main.rs` only. For the introduction of the default plugin configuration, I chose not to pass the `TEdgeConfig` object into the agent structs and keep that dependency limited to just an `Option<String>` for easier testability. Since it already lead me down that "minimal dependency" path, thought of just completing that cleanup by making these changes as well.